### ISSUE TITLE
[jupyterlab] [esrf] [jupyter-slurm] very slow reading and broken `.h5`?

### ISSUE THREAD

**Issue by jpcbertoldo** [silx-kit/jupyterlab-h5web#51]
Hi, I was using `jupyterlab-h5web` on [jupyter-slurm.esrf.fr](jupyter-slurm.esrf.fr), and some files are veeeery slow, and I have one that doesn't even open. I am not sure if this is somehow expected/normal, but I recorded a few of this issues so you can check it. 

See: [https://youtu.be/qSM2JsVVEHY](https://youtu.be/qSM2JsVVEHY)

I open 4 files in the video, respectively: 

1) [0:08: an `.h5` from the acquisition system at `id11`](https://youtu.be/qSM2JsVVEHY?t=8)
        path: `/data/id11/3dxrd/blc12852/id11/bmg_l1/bmg_l1_bmg_dct2/bmg_l1_bmg_dct2.h5`
	issue: [0:38: displaying one frame is quite slow](https://youtu.be/qSM2JsVVEHY?t=38)
	
2) [1:15: a "small" `.h5` written by my code is quite slow](https://youtu.be/qSM2JsVVEHY?t=75)
	path: `/data/id11/3dxrd/blc12852/id11/analysis/bmg_l1_bmg_dct2/difspots/000000.h5`
	issue: isn't it slow relative to the size of the things inside?

3) [1:33: an `.h5` regrouping (external links) about 3300 `.h5`s like the one above](https://youtu.be/qSM2JsVVEHY?t=93)
	path: `/data/id11/3dxrd/blc12852/id11/analysis/bmg_l1_bmg_dct2/difspots.h5`
	issue: slow to open, then very slow to expand the group with the external links

4) [2:30: an `.h5` with only 2 datasets, about the same size of the first one](https://youtu.be/qSM2JsVVEHY?t=150)
	path: `/data/id11/3dxrd/blc12852/id11/bmg_l1/bmg_l1_bmg_dct2/bmg_l1_bmg_dct2.preprocessed.h5`
	issue: veeeeeeeery slow; I kept waiting after the end of the video for several minutes, the entire `jupyterlab` becomes very slow, and it ends up breaking the page (screenshot below). When I reload the page, the `h5web` tab (in `jupyterlab`) is not there anymore.

![Screenshot from 2021-08-26 14-31-13](https://user-images.githubusercontent.com/24547377/130965603-bbddb71d-f010-453a-abf1-46eb44edb3c0.png)

A few notes (Idk if this might have some influence):
- I am using symbolic links (`ln -s`) to be able to see those files in `jupyter-slurm` because they are in `/data`.
- I am writting my `.h5` files with `silx.io.dictdump.dicttoh5`. I can show you specifically where in my code.
- I saw similar effects on a `jupyterlab` instances on `gpid11-nice` and `rnice8` nodes. In these cases I launch `jupyterlab` on those machines, connect to them with `ssh -L 8888:localhost:8888` to forward the application's port, then access it using my local browser (my personal computer, not an ESRF one).

**Comment by loichuder** [comment]
Hello @joaopcbertoldo and thanks for the feedback ! To add on @t20100 comments:

- Even though `jupyterlab-h5web` only officially supports JupyterLab **2.x**, CI tests should ensure that the extension should work for JupyterLab **3.0.x**. Note that so far (v0.0.9 and below), the extension does not work with JupyterLab **3.1.x** (see #49).
- Even if we release a version of `jupyterlab-h5web` compatible with JupyterLab **3.1.x** (which we will do!), this will not fix #24 straight away. There are still roadblocks on the way that I need to overcome by changing things in the extension and in core JupyterLab. That is why we put forward the opening of files in the notebook rather than in the browser.
- Now getting back to the issues you reported, I expect 4. to be due to #24 so there is not much to be done. 
- The other operations seem indeed slower than what I would expect so I will investigate. I will have a go with your files to see if I get the same.

**Comment by loichuder** [comment]
As already mentioned, **the root issue is that opening an h5web tab by double-clicking on a file fetched the whole file as a base64 blob**. Not only this is quite inefficient but also **pending requests of this type will slow down the whole "regular" fetching process**.

I thought we needed to wait for a newer version of JLab to remove this blob fetching but I was able to bypass the problematic machinery in 0.10.0. Therefore, **0.10.0 removed this blob fetching.** 

Getting back to the issue at hand: **0.10.0 is now deployed on jupyter-slurm** and I tried all the operations above. Except for the external links one (3.), **I found that operations were way faster than what you presented** so I would invite you to retry and tell us if you get the same impression.
